protocol-designer/src/step-forms/utils/index.ts

```diff
@@ -128,17 +143,22 @@ export const getSlotIsEmpty = (
     return false
   }
 
-  const filteredAdditionalEquipmentOnDeck = includeStagingAreas
-    ? values(
-        initialDeckSetup.additionalEquipmentOnDeck
-      ).filter((additionalEquipment: AdditionalEquipmentOnDeck) =>
-        additionalEquipment.location?.includes(slot)
-      )
-    : values(initialDeckSetup.additionalEquipmentOnDeck).filter(
-        (additionalEquipment: AdditionalEquipmentOnDeck) =>
-          additionalEquipment.location?.includes(slot) &&
-          additionalEquipment.name !== 'stagingArea'
-      )
+  const filteredAdditionalEquipmentOnDeck = values(
+    initialDeckSetup.additionalEquipmentOnDeck
+  ).filter((additionalEquipment: AdditionalEquipmentOnDeck) => {
+    const cutoutForSlotOt2 = slotToCutoutOt2Map[slot]
+    const includeStaging = includeStagingAreas
+      ? true
+      : additionalEquipment.name !== 'stagingArea'
+    if (cutoutForSlotOt2 != null) {
+      //  for Ot-2
+      return additionalEquipment.location === cutoutForSlotOt2 && includeStaging
+    } else {
+      //  for Flex
+      return additionalEquipment.location?.includes(slot) && includeStaging
```

Comment:
nit 
The following might be safe.  

```suggestion
      return ((additionalEquipment.location?.includes(slot)) ?? false) && includeStaging
```